src/stats/rawsender.cpp

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2017-2023 Vincent Thiery
+// Copyright (c) 2024 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <stats/rawsender.h>
+
+#include <netaddress.h>
+#include <netbase.h>
+#include <util/sock.h>
+#include <util/thread.h>
+
+RawSender::RawSender(const std::string& host, uint16_t port, std::pair<uint64_t, uint8_t> batching_opts,
+                     uint64_t interval_ms, std::optional<std::string>& error) :
+    m_host{host},
+    m_port{port},
+    m_batching_opts{batching_opts},
+    m_interval_ms{interval_ms}
+{
+    if (host.empty()) {
+        error = "No host specified";
+        return;
+    }
+
+    if (auto netaddr = LookupHost(m_host, /*fAllowLookup=*/true); netaddr.has_value()) {
+        if (!netaddr->IsIPv4()) {
+            error = strprintf("Host %s on unsupported network", m_host);
+            return;
+        }
+        if (!CService(*netaddr, port).GetSockAddr(reinterpret_cast<struct sockaddr*>(&m_server.first), &m_server.second)) {
+            error = strprintf("Cannot get socket address for %s", m_host);
+            return;
+        }
+    } else {
+        error = strprintf("Unable to lookup host %s", m_host);
+        return;
+    }
+
+    SOCKET hSocket = ::socket(AF_INET, SOCK_DGRAM, IPPROTO_UDP);
```

Comment:
try to use a helper CreateSockTCP from netbase.h instead re-writing low-level code

Reply:
`CreateSockTCP`, as the name suggests, is used for TCP connections. Currently Statsd messages are transmitted over UDP. One of the earlier drafts of #5167 did create a `CreateSockUDP` but it would not only make future backports affecting `netbase.{cpp,h}` painful, it was clunky from the get-go.

Furthermore, `Sock` only envisions the usage of TCP, providing only a wrapper for `::send()` (and not `::sendto()`, which is generally an applicable way to send messages when using UDP) ([source](https://github.com/dashpay/dash/blob/4c1e0348b399854c28fe92fa69b6bbc3feca4a35/src/util/sock.h#L130-L134))